File: backend/eventticketingplatform/src/main/java/com/nipapager/eventticketingplatform/event/service/TicketTypeService.java
package com.nipapager.eventticketingplatform.event.service;

import com.nipapager.eventticketingplatform.event.dto.TicketTypeDTO;
import com.nipapager.eventticketingplatform.response.Response;

import java.util.List;

/**
 * Service interface for ticket type operations
 */
public interface TicketTypeService {

    Response<TicketTypeDTO> createTicketType(Long eventId, TicketTypeDTO ticketTypeDTO);

    Response<List<TicketTypeDTO>> getTicketTypesByEventId(Long eventId);

    Response<TicketTypeDTO> getTicketTypeById(Long id);

    Response<TicketTypeDTO> updateTicketType(Long id, TicketTypeDTO ticketTypeDTO);

    Response<Void> deleteTicketType(Long id);
}

File: frontend/event-ticketing-app/src/api/ticketTypeService.ts
import api from './axios';

interface TicketType {
  id: number;
  eventId: number;
  eventName: string;
  name: string;
  price: number;
  totalQuantity: number;
  quantityAvailable: number;
  createdAt: string;
  updatedAt: string;
}

const ticketTypeService = {
  // Create ticket type for event
  createTicketType: async (eventId: number, ticketData: {
    name: string;
    price: number;
    totalQuantity: number;
  }): Promise<TicketType> => {
    const response = await api.post(`/ticket-types/event/${eventId}`, ticketData);
    return response.data.data;
  },

  // Get ticket types for event
  getTicketTypesByEvent: async (eventId: number): Promise<TicketType[]> => {
    const response = await api.get(`/ticket-types/event/${eventId}`);
    return response.data.data;
  },

  // Update ticket type
  updateTicketType: async (id: number, ticketData: {
    name?: string;
    price?: number;
    quantityAvailable?: number;
  }): Promise<TicketType> => {
    const response = await api.put(`/ticket-types/${id}`, ticketData);
    return response.data.data;
  },

  // Delete ticket type
  deleteTicketType: async (id: number): Promise<void> => {
    const response = await api.delete(`/ticket-types/${id}`);
    return response.data;
  }
};

export default ticketTypeService;